src/hyperlane-adapter/HyperlaneSenderAdapterV2.sol

```diff
@@ -58,10 +61,18 @@ contract HyperlaneSenderAdapter is ISingleMessageDispatcher, IBatchedMessageDisp
    * @notice HyperlaneSenderAdapter constructor.
    * @param _mailbox Address of the Hyperlane `Mailbox` contract.
    */
-  constructor(address _mailbox, address _igp) {
+  constructor(
+    address _mailbox,
+    address _igp,
+    uint256 _gasAmount
+  ) {
     if (_mailbox == address(0)) {
       revert Errors.InvalidMailboxZeroAddress();
     }
+
+    // See https://docs.hyperlane.xyz/docs/build-with-hyperlane/guides/paying-for-interchain-gas
+    // Set gasAmount to the default (500,000) if _gasAmount is 0
+    gasAmount = (_gasAmount == 0) ? 500000 : _gasAmount;
```

Comment:
made gasAmount a configurable option in the constructor, it it isn't set (set to zero), it defaults to 500k